### PR TITLE
fix: exception when recording with no tests in specfile

### LINKED ISSUE #15512: Error when running test suite with empty spec files
### Current behavior

I upgraded to Cypress 6.7.1 and now [there's an error](https://github.com/tutorbookapp/tutorbook/runs/2118940428?check_suite_focus=true#step:9:42) when using `cypress run` with empty spec files:

```
TypeError: Cannot read property 'tests' of null
    at Object.exports.flattenSuiteIntoRunnables (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/util/tests_utils.js:12:32)
    at ProjectE2E.onTestsReceived [as _recordTests] (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/modes/record.js:727:30)
    at ProjectE2E.<anonymous> (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/project-base.js:280:32)
    at Generator.next (<anonymous>:null:null)
    at /home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/project-base.js:9:71
    at new Promise (<anonymous>:null:null)
    at __awaiter (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/project-base.js:5:12)
    at Object.onTestsReceivedAndMaybeRecord (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/project-base.js:274:63)
    at SocketE2E.<anonymous> (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/socket-base.js:222:32)
    at Generator.next (<anonymous>:null:null)
    at /home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/socket-base.js:8:71
    at new Promise (<anonymous>:null:null)
    at __awaiter (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/socket-base.js:4:12)
    at Socket.<anonymous> (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/server/lib/socket-base.js:221:82)
    at Socket.emit (events.js:315:20)
    at Socket.onevent (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/socket.js:253:20)
    at Socket._onpacket (/home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/socket.js:216:22)
    at /home/runner/.cache/Cypress/6.7.1/Cypress/resources/app/packages/socket/node_modules/socket.io/dist/client.js:205:28
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
```

### Desired behavior

There should be no error when running empty spec files. Instead, Cypress should just log a warning that they're empty and move on.

### Test code to reproduce

This should be relatively simple to reproduce.:
1. Just create a new Cypress repo
2. Add an empty test
3. And run Cypress

Here's [the link to the original repository](https://github.com/tutorbookapp/tutorbook) where I'm seeing the error.

### Versions

Cypress: 6.7.1
Browser: N/A

### PR DESCRIPTION
**Pull request by kuceb**
- Closes #15512 

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue or this PR been tagged with a release in ZenHub?

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
Keep getting this error when trying to record in staging. 

```
Recording this run failed because the request was invalid.

Request should follow postRunRequest@2.2.0 schema

Errors:

[
  "data has additional properties: runnerCapabilities"
]

Request Sent:

{
  "ci": {
    "params": null,
    "provider": null
  },
  "specs": [
    "cypress/integration/spec.js",
    "cypress/integration/untitled_spec.js",
    "cypress/integration/untitled_spec2.js",
    "cypress/integration/untitled_spec3.js"
  ],
  "commit": {
    "sha": "71b5c34e5b4bb24fd00a9e43a4222c31161299c1",
    "branch": "master",
    "authorName": "Ralf Richtsteiger",
    "authorEmail": "dev@rwam.de",
    "message": "add example code\n",
    "remoteOrigin": "https://github.com/rwam/cypress-test.git",
    "defaultBranch": null
  },
  "group": null,
  "platform": {
    "osCpus": [
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 48215720,
          "nice": 0,
          "sys": 31910740,
          "idle": 144684890,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 3793580,
          "nice": 0,
          "sys": 5562250,
          "idle": 215287340,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 41126270,
          "nice": 0,
          "sys": 23321200,
          "idle": 160199290,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 3782870,
          "nice": 0,
          "sys": 5308350,
          "idle": 215551590,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 34483520,
          "nice": 0,
          "sys": 19475270,
          "idle": 170687320,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 3794280,
          "nice": 0,
          "sys": 5127750,
          "idle": 215720400,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 28699140,
          "nice": 0,
          "sys": 16052610,
          "idle": 179893690,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 3790350,
          "nice": 0,
          "sys": 4994960,
          "idle": 215856750,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 25706030,
          "nice": 0,
          "sys": 14136700,
          "idle": 184802050,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 3758100,
          "nice": 0,
          "sys": 4879550,
          "idle": 216004010,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 21328750,
          "nice": 0,
          "sys": 11575640,
          "idle": 191739730,
          "irq": 0
        }
      },
      {
        "model": "Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz",
        "speed": 2600,
        "times": {
          "user": 3746870,
          "nice": 0,
          "sys": 4782160,
          "idle": 216112250,
          "irq": 0
        }
      }
    ],
    "osName": "darwin",
    "osMemory": {
      "free": 842321920,
      "total": 17179869184
    },
    "osVersion": "19.6.0",
    "browserName": "Electron",
    "browserVersion": "87.0.4280.141"
  },
  "parallel": null,
  "ciBuildId": null,
  "projectId": "6wa698",
  "recordKey": "5df12e37-11cc-4f1d-a0a0-a7e6842aef6e",
  "specPattern": null,
  "tags": [
    ""
  ],
  "runnerCapabilities": {
    "dynamicSpecsInSerialMode": true,
    "skipSpecAction": true
  }
}
```

**Review by jennifer-shehane — APPROVED** [review]
Manually tested - this fixes the issue. Also tested.

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
This exits and does report a Dashboard url, but it looks like it reports as pending forever. So, not sure if this is relevant. https://dashboard-staging.cypress.io/projects/o9ej5p/runs/1

**Comment by brian-mann** [comment]
This sounds pretty relevant to me... @Bkucera and @traviscy thoughts?

**Review by brian-mann — DISMISSED** [review]

File: packages/server/test/e2e/7_record_spec.js [hunk 1]
```diff
@@ -406,6 +406,35 @@ describe('e2e record', () => {
     })
   })
 
+  context('empty specs', () => {
+    setupStubbedServer(createRoutes())
+
+    it('succeeds when empty spec file', async function () {
```
Comment (brian-mann) [inline packages/server/test/e2e/7_record_spec.js:412]: you should link to the github issue that this is fixing, this is a specific regression

File: packages/server/lib/util/tests_utils.ts [hunk 2]
```diff
@@ -9,6 +9,10 @@ export const flattenSuiteIntoRunnables = (suite, tests = [], hooks = []) => {
     )
   }
 
+  if (!suite || !suite.suites) {
```
Comment (brian-mann) [inline packages/server/lib/util/tests_utils.ts:12]: maybe explain why this is necessary and link to the github issue that this is fixing, this is a specific regression

**Pull request merged**